packages/experiments-realm/url.gts

```diff
@@ -0,0 +1,66 @@
+import {
+  StringField,
+  Component,
+  field,
+  CardDef,
+  contains,
+} from 'https://cardstack.com/base/card-api';
+import { BoxelInput } from '@cardstack/boxel-ui/components';
+import { not } from '@cardstack/boxel-ui/helpers';
+
+import ExternalLink from '@cardstack/boxel-icons/external-link';
+
+export class UrlField extends StringField {
+  static icon = ExternalLink;
+  static displayName = 'Url';
+
+  static edit = class Edit extends Component<typeof UrlField> {
+    get isValidUrl() {
+      if (!this.args.model) {
+        return false;
+      }
+      return isValidUrl(this.args.model);
+    }
+    <template>
+      <BoxelInput
+        type='url'
+        value={{@model}}
+        @onInput={{@set}}
+        @disabled={{not @canEdit}}
+        @state={{if this.isValidUrl 'valid' 'invalid'}}
+      />
+    </template>
+  };
+
+  static atom = class Atom extends Component<typeof UrlField> {
+    <template>
+      {{#if @model}}
+        {{#if (isValidUrl @model)}}
+          <a href={{@model}}>{{@model}}</a>
+        {{else}}
+          Invalid URL
+        {{/if}}
+      {{/if}}
+    </template>
+  };
+}
+
+function isValidUrl(urlString: string): boolean {
+  try {
+    new URL(urlString);
+    return true;
+  } catch (err) {
+    return false;
+  }
```

Comment:
`URL.canParse` [does exist](https://caniuse.com/mdn-api_url_canparse_static) for this use.

Reply:
In order for this to work with types, I'd need to upgrade to @typescript 5.2. Shall I upgrade our types to the most recent version of typescript or keep it at 5.2.2? I will merge first and resolve in another PR.

https://github.com/microsoft/TypeScript/issues/55623#issuecomment-1704705924